pkg/osvscanner/vulnerability_result.go

```diff
@@ -74,3 +75,58 @@ func groupResponseBySource(r reporter.Reporter, query osv.BatchedQuery, resp *os
 
 	return output
 }
+
+// diffVulnerabilityResults will return any new vulnerabilities that are in `newRes`
+// which is not present in `oldRes`, but not the reverse.
+//
+// Current implementation is O(n^2) on the number of vulns, but can be reduced to linear time
+func diffVulnerabilityResults(oldRes, newRes models.VulnerabilityResults) models.VulnerabilityResults {
+	result := models.VulnerabilityResults{}
+	for _, ps := range newRes.Results {
+		sourceIdx := slices.IndexFunc(oldRes.Results, func(elem models.PackageSource) bool { return elem.Source == ps.Source })
+		if sourceIdx == -1 {
+			result.Results = append(result.Results, ps)
+			continue
+		}
+		result.Results = append(result.Results, models.PackageSource{
+			Source: ps.Source,
+		})
+		resultPS := &result.Results[len(result.Results)-1]
+		for _, pv := range ps.Packages {
+			pkgs := oldRes.Results[sourceIdx].Packages
+			pkgIdx := slices.IndexFunc(pkgs, func(elem models.PackageVulns) bool { return elem.Package == pv.Package })
+			if pkgIdx == -1 {
+				resultPS.Packages = append(resultPS.Packages, pv)
```

Comment:
Add a comment: "Newly introduced package vulnerability".

Reply:
Done in: 5f1fd7a471c91a6065a92814389ab6399a399bee